### DISCUSSION TITLE
Update database in one call to prisma. Conditional increment / decrement? Thanks in advance!

### DISCUSSION THREAD

**Discussion by mah51** [prisma/prisma#13555] [Q&A]
Hi I am looking for a way to do this whole operation in one update() call:

```
const updatedGame = await container.prisma.game.update({
        where: {
            id: game.id
        },
        data: {
            state: 'END',
            result: areTeamOneWinners ? 1 : 2,
            mmrAwarded,
            players: {
                updateMany: {
                    where: {
                        id: {
                            in: areTeamOneWinners ? game.team1 : game.team2
                        }
                    },
                    data: {
                        winCount: {
                            increment: 1
                        },
                        mmr: {
                            increment: mmrAwarded
                        }
                    }
                }
            }
        }
    });

    await container.prisma.member.updateMany({
        where: {
            id: {
                in: areTeamOneWinners ? game.team2 : game.team1
            }
        },
        data: {
            lossCount: {
                increment: 1
            },
            mmr: {
                decrement: mmrAwarded
            }
        }
    });

```
I find the game and add the mmr to all winners in the game but then have to do a search for all losers and decrease their mmr in another one, is there any way to decrease the mmr in the first update call?

**Comment by nurul3101** [comment]
Hey @mah51 👋 
I would recommend doing both of these operations in a [transaction](https://www.prisma.io/docs/concepts/components/prisma-client/transactions). It allows you to perform a single Prisma Client API call with multiple operations that touch multiple [related](https://www.prisma.io/docs/concepts/components/prisma-schema/relations) records and ensures that all operations succeed or fail as a whole.

Also for your question of if it's possible to decrease mmr in first update call, could you share the models or your `schema.prisma` file?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
